integrations/weaviate/pyproject.toml

```diff
@@ -28,6 +26,7 @@ dependencies = [
   "haystack-ai",
   "weaviate-client==3.*",
   "haystack-pydoc-tools",
+  "python-dateutil",
```

Comment:
I had to add this to have proper parsing of ISO 8601 dates as [`datetime.fromisoformat()`](https://docs.python.org/3/library/datetime.html#datetime.date.fromisoformat) supports it properly only from `3.11`.